msg/templates/uorb/uORBTopics.hpp.em

```diff
@@ -0,0 +1,76 @@
+@###############################################
+@#
+@# EmPy template for generating uORBTopics.hpp file
+@# for logging purposes
+@#
+@###############################################
+@# Start of Template
+@#
+@# Context:
+@#  - msgs (List) list of all msg files
+@#  - multi_topics (List) list of all multi-topic names
+@#  - ids (List) list of all RTPS msg ids
+@###############################################
+/****************************************************************************
+ *
+ *   Copyright (C) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+@{
+msg_names = [mn.replace(".msg", "") for mn in msgs]
+msgs_count = len(msg_names)
+msg_names_all = list(set(msg_names + multi_topics)) # set() filters duplicates
+msg_names_all.sort()
+msgs_count_all = len(msg_names_all)
+}@
+
+#pragma once
+
+#include <stddef.h>
+
+#include <uORB/uORB.h>
+
+static constexpr size_t ORB_TOPICS_COUNT{@(msgs_count_all)};
+static constexpr size_t orb_topics_count() { return ORB_TOPICS_COUNT; }
+
+/*
+ * Returns array of topics metadata
+ */
+extern const struct orb_metadata *const *orb_get_topics() __EXPORT;
+
+enum class ORB_ID : uint8_t {
```

Comment:
Won't this cause troubles with the `ORB_ID` macro? At the least it will be confusing.

Reply:
We can get away with it because of the parentheses, but I'm not sure how questionable it is. The idea is make them equivalent and interchangeable in most uORB usage and possibly transition entirely to the enum class version later.

Reply:
If we were ready/willing to drop orb C support entirely we could actually get rid of the ORB_ID macro as a wrapper for  orb meta pointer (eg `__orb_vehicle_status`) and replace it with a single byte enum class end-to-end.

Then `ORB_ID(vehicle_status)` would simply give you `ORB_ID::vehicle_status`.